src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -1,171 +1,560 @@
+/**
+ * Helper function to load scripts.
+ *
+ * It runs only once, so it won't load the same script on this or any other visualization show/hide.
+ */
 function loadScript(url) {
-    var script = document.createElement("script");
+    let script = document.createElement("script");
     script.src = url;
 
     document.head.appendChild(script);
 }
 
-loadScript('https://d3js.org/d3.v4.min.js');
+/**
+ * Helper function to load styles.
+ *
+ * It runs only once, so it won't load the same style on this or any other visualization show/hide.
+ */
+function loadStyle(url) {
+    let style   = document.createElement("link");
+    style.href  = url;
+    style.rel   = "stylesheet";
+    style.media = "screen";
+    style.type  = "text/css";
+
+    document.head.appendChild(style);
+}
```

Comment:
Where in the code it is guaranteed it runs only once? I see we call it on every incoming data, where is the check?

Reply:
No. The code is run once per loading of visualization. On every incoming data the onDataReceived is called.

Reply:
Ditto: Actually the "before" implementation worked perfectly because the file was loaded only once, and then, the only thing that got called was the onDataReceived. So by implementation it is and must be load only once.

Reply:
Ok, that makes sense, but now I dont understand the docs here. The docs here tell that it runs only once, but this is not guaranteed in the body. So the docs are wrong now.